test/unit/verifier/lint/test_precommit.py

```diff
@@ -0,0 +1,184 @@
+#  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+#  sell copies of the Software, and to permit persons to whom the Software is
+#  furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included in
+#  all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+#  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+#  DEALINGS IN THE SOFTWARE.
+
+import pytest
+import sh
+
+from molecule import config
+from molecule.verifier.lint import precommit
+
+
+@pytest.fixture
+def _patched_get_tests(mocker):
+    m = mocker.patch('molecule.verifier.lint.precommit.PreCommit._get_tests')
+    m.return_value = ['test1', 'test2', 'test3']
+
+    return m
+
+
+@pytest.fixture
+def _verifier_lint_section_data():
+    return {
+        'verifier': {
+            'name': 'testinfra',
+            'lint': {
+                'name': 'pre-commit',
+                'options': {
+                    'foo': 'bar',
+                },
+                'env': {
+                    'FOO': 'bar',
+                },
+            }
+        }
+    }
+
+
+# NOTE(retr0h): The use of the `patched_config_validate` fixture, disables
+# config.Config._validate from executing.  Thus preventing odd side-effects
+# throughout patched.assert_called unit tests.
+@pytest.fixture
+def _instance(patched_config_validate, config_instance):
+    return precommit.PreCommit(config_instance)
+
+
+def test_config_private_member(_instance):
+    assert isinstance(_instance._config, config.Config)
+
+
+def test_default_options_property(_instance):
+    assert {} == _instance.default_options
+
+
+def test_default_env_property(_instance):
+    assert 'MOLECULE_FILE' in _instance.default_env
+    assert 'MOLECULE_INVENTORY_FILE' in _instance.default_env
+    assert 'MOLECULE_SCENARIO_DIRECTORY' in _instance.default_env
+    assert 'MOLECULE_INSTANCE_CONFIG' in _instance.default_env
+
+
+@pytest.mark.parametrize(
+    'config_instance', ['_verifier_lint_section_data'], indirect=True)
+def test_env_property(_instance):
+    assert 'bar' == _instance.env['FOO']
+
+
+@pytest.mark.parametrize(
+    'config_instance', ['_verifier_lint_section_data'], indirect=True)
+def test_name_property(_instance):
+    assert 'pre-commit' == _instance.name
+
+
+def test_enabled_property(_instance):
+    assert _instance.enabled
+
+
+@pytest.mark.parametrize(
+    'config_instance', ['_verifier_lint_section_data'], indirect=True)
+def test_options_property(_instance):
+    x = {
```

Comment:
This is totally optional BUT if you want to use more useful naming, that would be great :)

Reply:
I changed the options to be more realistic for pre-commit. (disable color output).

Reply:
It was about naming X with something more meaningful. What does it refer to?

Reply:
This one is fixed too.